### DISCUSSION TITLE
Boost Event

### DISCUSSION THREAD

**Discussion by veni-vidi-code** [Rapptz/discord.py#7146] [Q&A]
Is there a possibility to create an event when someone boosts the server?

**Comment by Bluenix2** [comment]
I do not believe Discord sends a separate event when this happens, but you should be able to listen to `on_message` and figure out when Discord sends one of those "Bluenix has boosted the server, it reached level 2!" messages.

**Comment by Tari-dev** [comment]
When a server is boosted, `on_guild_update` is dispatched afaik.
Since `on_guild_update` is dispatched for many other event as well, u need to check if Guild.premium_subscription_count has increased.

**Reply by veni-vidi-code** [reply]
Thanks, this helps

**Reply by AnotherTwinkle** [reply]
This isn't the ideal solution as `on_guild_update` doesn't dispatch the member associated with that event if any. Checking for guild boosts should be done with `on_member_update`. (Although this is only applicable for the first boost.)

**Reply by Tari-dev** [reply]
There is no ideal solution for boost event since discord doesn't want to provide information regarding nitro and nitro boosters.
Further more, You can get the member by comparing `Guild.premium_subscribers` in `on_guild_update` (Again this only works if member is boosting for the first time)

Just like bluenix mentioned, the solution is to listen to [system messages.](https://discordpy.readthedocs.io/en/master/api.html#discord.MessageType.premium_guild_subscription)  (note that system boost messages are optional)
